code/modules/admin/verbs/adminhelp.dm

```diff
@@ -938,65 +938,74 @@ GLOBAL_DATUM_INIT(admin_help_ui_handler, /datum/admin_help_ui_handler, new)
+ * 2. Datum refs, like @0x2001169 or @mob_23. Clicking on the link opens up the VV for that datum
+ * 3. Ticket refs, like #3. Displays the status and ahelper in the link, clicking on it brings up the ticket panel for it.
+ * Returns a list being used as a tuple. Index ASAY_LINK_NEW_MESSAGE_INDEX contains the new message text (with clickable links and such)
+ * while index ASAY_LINK_PINGED_ADMINS_INDEX contains a list of pinged admin clients, if there are any.
  *
  * Arguments:
  * * msg - the message being scanned
  */
-/proc/check_admin_pings(msg)
-	//explode the input msg into a list
-	var/list/msglist = splittext(msg, " ")
-	var/list/admins_to_ping = list()
+/proc/check_asay_links(msg)
+	var/list/msglist = splittext(msg, " ") //explode the input msg into a list
+	var/list/pinged_admins = list() // if we ping any admins, store them here so we can ping them after
+	var/modified = FALSE // did we find anything?
 
 	var/i = 0
 	for(var/word in msglist)
 		i++
 		if(!length(word))
 			continue
-		if(word[1] != "@")
-			continue
-		var/ckey_check = lowertext(copytext(word, 2))
-		var/client/client_check = GLOB.directory[ckey_check]
-		if(client_check?.holder)
-			msglist[i] = "<u>[word]</u>"
-			admins_to_ping[ckey_check] = client_check
 
-	if(length(admins_to_ping))
-		admins_to_ping[ADMINSAY_PING_UNDERLINE_NAME_INDEX] = jointext(msglist, " ") // without tuples, we must make do!
-		return admins_to_ping
-
-/**
- * Checks a given message to see if any of the words contain a memory ref for a datum. Said ref should not have brackets around it
- *
- * Returns nothing if no refs are found, otherwise returns an associative list with ckey -> client
- * Also modifies msg to underline and linkify the [ref] so other admins can click on the address to open the VV entry for said datum
- *
- * Arguments:
- * * msg - the message being scanned
- */
-/proc/check_memory_refs(msg)
-	if(!findtext(msg, GLOB.is_memref))
-		return
-
-	//explode the input msg into a list
-	var/list/msglist = splittext(msg, " ")
-	var/list/datums_to_ref = list()
-
-	var/i = 0
-	for(var/word in msglist)
-		i++
-		if(!length(word))
-			continue
-		var/word_with_brackets = "\[[word]\]" // the actual memory address lookups need the bracket wraps
-		var/datum/check_datum = locate(word_with_brackets)
-		if(!istype(check_datum))
-			continue
-		msglist[i] = "<u><a href='?_src_=vars;[HrefToken(TRUE)];Vars=[word_with_brackets]'>[word_with_brackets]</A></u>"
-		datums_to_ref[word] = word
+		switch(word[1])
+			if("@")
+				var/stripped_word = lowertext(copytext(word, 2))
+
+				// first we check if it's a ckey of an admin
+				var/client/client_check = GLOB.directory[stripped_word]
+				if(client_check?.holder)
+					msglist[i] = "<u>[word]</u>"
+					pinged_admins[stripped_word] = client_check
+					modified = TRUE
+					continue
+
+				// then if not, we check if it's a datum ref
+
+				var/word_with_brackets = "\[[stripped_word]\]" // the actual memory address lookups need the bracket wraps
```

Comment:
don't need the \] at the end here

Reply:
It's intentional, so that it shows "[tag]", matches the opening escaped bracket